flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = "Dev environment for jank";
+
+  inputs = {
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs = inputs @ {flake-parts, ...}:
+    flake-parts.lib.mkFlake {inherit inputs;} {
+      systems = ["x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin"];
+      perSystem = {
+        pkgs,
+        ...
+      }: {
+        legacyPackages = pkgs;
+        formatter = pkgs.alejandra;
+        devShells.default = (pkgs.mkShell.override { stdenv = pkgs.llvmPackages.stdenv; }) {
+          packages = with pkgs; [
+            # Build deps.
+            cmake
+            ninja
+            pkg-config
+            clang
+
+            # Dev tools.
+            entr
+            gcovr
+            lcov
+            git
+            nixd
+            shellcheck
+            # For clangd
+            llvm
+            llvmPackages.libclang
+
+            # Libs.
+            boehmgc
+            boost
+            cli11
+            fmt
+            immer
+            libzip
+            magic-enum
+            openssl
+            readline
```

Comment:
I think we can strip this down.

* `immer`, `fmt`, and `cli11` are all built along with jank as submodules, so should not be needed
* `magic-enum` should not be needed at all
* `readline` should not be needed at all

Reply:
These were just copied from the old `shell.nix`. I've removed them